src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,39 +41,73 @@
  * {@link org.springframework.batch.item.data.MongoItemReader} is using
  * pagination and it is slow with large collections
  */
-public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
+public class NonAnnotatedVariantsMongoReader implements ItemStreamReader<VariantWrapper>, InitializingBean {
```

Comment:
Specifically, what convolution are you referring to? The structure of this class is similar to a decorator pattern: we have the generic MongoDbCursorItemReader, and the AnnotatedVariantsMongoReader narrows the goal to read VariantWrappers, and for that it needs a specific query and a specific conversion (and the latter is already contained in the DBObjectToVariantConverter)